MIGRATION.md

```diff
@@ -35,6 +45,88 @@ You can only run one codemod at a time and we encourage you to apply the transfo
 
+> ⚠️ If you run into `'node\r': No such file or directory` when running the codemods with yarn, run them with node directly instead (this is a [known issue](https://github.com/facebook/jscodeshift/issues/424)).
+>
+> ```sh
+> ./node_modules/.bin/circuit-ui migrate -l JavaScript -l TypeScript -t codemod-name
+> ```
+
+## From v2.x to v3
+
+(intro)
+
+### JSX runtime
+
+(...)
+
+### Typography XXX
+
+Before v3, typography components were general purpose and flexible. There were no guidelines on when to use a certain typography style. This has led to inconsistent usage in our apps.
+
+The new version introduces new, semantic typography components that make it clear when each should be used, are flexible enough to cover all use cases, and for the first time include semantic colors as well.
+
+#### Typography component names
+
+The core typography components were renamed:
+
+| v2           | v3                                                     |
+| ------------ | ------------------------------------------------------ |
+| `Text`       | `Body`                                                 |
+| `Heading`    | `Headline`                                             |
+| `SubHeading` | `SubHeadline`                                          |
+| `Blockquote` | See the [next section](#typography-component-variants) |
+
+(🤖 _component-names-v3_.)
+
+> _Note: this codemod also transforms other renamed components (see [Other Changes](#other-changes)._
+
+#### Typography component variants
+
+The `Body` (formerly `Text`) component's variants were changed:
+
+| v2              | v3                           |
+| --------------- | ---------------------------- |
+| `<Text bold>`   | `<Body variant="highlight">` |
+| `<Text italic>` | Use custom styles            |
+| `<Text strike>` | Use custom styles            |
+| `<Blockquote>`  | `<Body variant="quote">`     |
+
+Use 🤖 _body-variant-highlight_ to migrate `bold` to `variant="highlight"`. Note that this will transform your `<p>` elements into inline `<strong>` elements: you might also need to pass `as="p"` if you need a block-level element.
+
+> Bonus: the new `Body` component also supports the `success`, `error` and `subtle` semantic variants. Have a look at the [story](https://circuit.sumup.com/?path=/story/typography-body--variants) to get started.
+
+#### New sizes
+
+(...)
+
+- Components: the `size` prop has been changed to accept the new size numbers for **Headline**, **SubHeadline**, **Body**. For **Headline** component **_exa_** and **_peta_** has been changed to **_one_** with new values, **_tera_** has been changed to **_two_**, **_giga_** to **_three_**, **_mega_** and **_kilo_** to **_four_**. For **Body** component **_giga_** size has been removed, and **_mega_** and **_kilo_** sizes have been changed to **_one_** and **_two_** respectively. The **SubHeadline** component now uses only one size value (🤖 _typography-sizes_).
```

Comment:
I wonder if this would be more digestible as a table.

Reply:
Yes, it will be one :slightly_smiling_face: Sorry this is still very much a WIP :construction: